ui/desktop/electron-app/src/services/client-daemon-manager.js

```diff
@@ -109,23 +109,41 @@ class ClientDaemonManager {
 }
 
 const addTokenCliCommand = (token) => {
-  const addTokenCommand = [
-    'daemon',
-    'add-token',
+  const sanitizedToken = sanitizer.base62EscapeAndValidate(token);
+  // The format of Boundary tokens is
+  //   token-id_token-secret
+  // where token-id has the format
+  //   at_1234567890,
+  // so we extract the token id by finding the last '_' and taking everything before it.
+  lastUnderscoreIndex = sanitizedToken.lastIndexOf('_');
+  if (lastUnderscoreIndex === -1) {
+    return Promise.reject({
+      statusCode: 400,
+      message: 'Invalid token format',
+    });
+  }
+  const tokenId = sanitizedToken.substring(0, lastUnderscoreIndex);
+  // Successfully calling any Boundary CLI command with a token
+  // will add the token both to the Client daemon and the Ferry DNS daemon,
+  // so we just do a simple read on the input token and let the CLI do the rest.
```

Comment:
I didn't realize calling any CLI command with a token automatically handles adding tokens to daemons, was this a recent change?

Also, is the recommendation to just go through the CLI for adding tokens at this point since it handles adding multiple tokens? I had a [TODO](https://github.com/hashicorp/boundary-ui/pull/2203/files#diff-3a31bb3a8a9063bb4dc5b0c1dba204044a1bc4543a264758d7cbd0962f195207R76) regarding whether we should just always go through the CLI for adding tokens and it sounds like it might be better at this point? This will only add the token right now for windows since it's the only one that calls the CLI directly (mac/linux adds the token to the client daemon directly right now).

If you prefer, I can update this PR if that's the recommended way.

Reply:
Automatically adding the token to the client daemon on any command was shipped with search and filtering - we definitely could've communicated that better, sorry! On Boundary `main` right now it will also add the token to the ferry daemon. I'd say going through the CLI is probably the simplest thing for all platforms, yeah.

One caveat I do want to mention is that ferry can be configured to serve on a different port than the default, and in these cases a separate flag will need to be set to let the CLI know where to direct its request (`--ferry-port`). I think we're OK with not supporting that until such a time as we have a way for the user to configure that in their Desktop client, but it's something to keep in mind.

If you'd like to commandeer this PR to make those changes, that'd be great! We only really care about it for Mac for now, since we're not shipping transparent sessions for Linux right now, but it can't hurt :).

Reply:
Sure I can do that, really appreciate you opening this PR! And thanks for the info, this is all good to know. 

Just for my edification, does the ferry service also expose its API over a unix socket like the client daemon?